ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,210 @@
+1.a) Example - Malicious scripts can detect when a cryptocurrency wallet address is copied to clipboard and replace it with attacker's address and that way redirect payment to attacker's account.  
+1.b) Example - Passwords / authentication keys / access tokens are frequently copied to clipboard and can be retrieved by malicious scripts reading clipboard in background.
+3. Can simplify implementation since browsers can simply check for clipboard change on page focus. However, this needs to be further investigated.
+4. This is in-line with current async clipboard focus APIs which require focus to access.
+
+##### Cons
+1. Might restrict web app scenarios which need to listen to clipboardchange events in the background.
+2. Could result in a less responsive user experience if clipboard changes are detected with a delay - if clipboard got changed when the browser was in background, the event is fired only when the browser regains focus. The delay here is the duration between actual copy of contents to clipboard and firing of the clipboardchange event in browser.
+
+#### 5.2.2 Approach 2 - No focus requirement
+
+##### Pros:
+1. Opens possibility for more user scenarios - e.g. For example, a web app can make network calls in the background when a specific clipboard item is updated - before pasting an image, a web app might want to check the image for embedded malicious content using a remote service. If the app can be notified about a clipboard change in background, it can trigger the necessary network operations while the user is in another page or application. When the user returns, the web app is ready with the results, reducing the wait time.
+
+##### Cons:
+1. Might be open to misuse - a web app will be able to monitor the clipboard even when the user is interacting with other applications / pages.
+2. Not much useful unless the page focus requirement is removed from the async read/write clipboard API.
+3. Could lead to higher resource consumption due to continuous monitoring. E.g. if a large number of pages are listening to the clipboardchange event, when the clipboard is changed, then all the pages will receive the event at once, which might put load on the system.
+
+#### 5.2.3 Conclusion
+We favour Approach 1 - Page required to be in focus to receive event, since this approach reduces the possibility of misusing the clipboard change event in privacy related attacks.
+
+### 5.3 Event details 
+"clipboardchange" is a system level event, hence it is not bound to any DOM element. Therefore, there is no bubbling associated with the event. This event simply signifies that the system clipboard got changed. Hence performing “preventDefault” for this event has no effect since there are no actions associated with it. The event does not provide any additional arguments in the callback handler - it is expected that the web app calls Async API to read the clipboard and get the changed clipboard contents within the callback.
```

Comment:
What does "system level" event mean? Is that a spec'ed term?

Reply:
"it is expected that the web app calls Async API to read the clipboard and get the changed clipboard contents within the callback" -> Is it at all possible to use the `clipboardchange` event with the DataTransfer API? If it is, we should mention how that could work with pros and cons, and justifying why the async clipboard API is recommended. If it isn't, perhaps we should mention that explicitly.

Reply:
Removed the term system level and clarified that it is not bound to any DOM element. Added details about DataTransfer API